server/src/test/java/org/elasticsearch/index/bulk/stats/BulkStatsTests.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.bulk.stats;
+
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.test.AbstractWireSerializingTestCase;
+
+public class BulkStatsTests extends AbstractWireSerializingTestCase<BulkStats> {
+
+    @Override
+    protected Writeable.Reader<BulkStats> instanceReader() {
+        return BulkStats::new;
+    }
+
+    @Override
+    protected BulkStats createTestInstance() {
+        return new BulkStats(randomNonNegativeLong(), randomNonNegativeLong(), randomNonNegativeLong());
+    }
+
+    @Override
+    protected BulkStats mutateInstance(BulkStats instance) {
+        BulkStats mutateBulkStats = new BulkStats(randomNonNegativeLong(), randomNonNegativeLong(), randomNonNegativeLong());
+        switch (between(0, 1)) {
+            case 0:
+                break;
+            case 1:
+                mutateBulkStats.add(instance);
+                break;
+            default:
+                throw new AssertionError("Illegal randomisation branch");
+        }
+        return mutateBulkStats;
+    }
```

Comment:
This isn't mutating the instance in the right way, here's a better mutation:

```java
protected BulkStats mutateInstance(BulkStats instance) {
  switch (between(0, 2)) {
    case 0:
      return new BulkStats(randomValueOtherThan(instance.getTotalOperations(), ESTestCase::randomNonNegativeLong),
          instance.getTotalTimeInMillis(), instance.getTotalSizeInBytes());
    case 1:
      return new BulkStats(instance.getTotalOperations(),
          randomValueOtherThan(instance.getTotalTimeInMillis(), ESTestCase::randomNonNegativeLong),
          instance.getTotalSizeInBytes());
    case 2:
      return new BulkStats(instance.getTotalOperations(), instance.getTotalTimeInMillis(),
          randomValueOtherThan(instance.getTotalSizeInBytes(), ESTestCase::randomNonNegativeLong));
    default:
      throw new AssertionError("Illegal randomisation branch");
  }
}
```

Where each branch mutates only a single value, and we ensure that we don't randomly get the same value by unlucky chance and cause a test failure.

(I didn't test this code, it might have a syntax error, it's just an example)